File: app/src/main/java/com/example/linqing/zhihumvp/detail/DetailPresenter.java
package com.example.linqing.zhihumvp.detail;

import android.content.Intent;
import android.webkit.WebView;

import com.example.linqing.zhihumvp.entitybean.ArticleDetailBean;


import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class DetailPresenter implements DetailContract.DetailPresenterInter {

    private DetailContract.DetailView detailView;
    private DetailModel detailModel = new DetailModel();
    String id;




    public DetailPresenter(DetailContract.DetailView detailView) {
        this.detailView = detailView;
    }

    @Override
    public void getIdP(String id){
        this.id=id;

    }



    @Override
    public void getWebviewData() {

        Call<ArticleDetailBean> call = detailModel.getId(id);
        call.enqueue(new Callback<ArticleDetailBean>() {

            @Override
            public void onResponse(Call<ArticleDetailBean> call, Response<ArticleDetailBean> response) {


                ArticleDetailBean data = response.body();
                detailView.loadWebDetail(data);


            }

            @Override
            public void onFailure(Call<ArticleDetailBean> call, Throwable t) {
                t.printStackTrace();
            }
        });
    }
}

File: app/src/main/java/com/example/linqing/zhihumvp/detail/DetailPresenter.kt
package com.example.linqing.zhihumvp.detail


import com.example.linqing.zhihumvp.entitybean.ArticleDetailBean
import kotlinx.coroutines.experimental.android.UI
import kotlinx.coroutines.experimental.launch
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.io.IOException

class DetailPresenter(private val detailView: DetailContract.DetailView) : DetailContract.DetailPresenterInter {
    private val detailModel = DetailModel()
    lateinit private  var id: String

    override fun getIdP(id: String) {
        this.id = id

    }


    override fun getWebviewData() {

        val call = detailModel.getId(id)
        launch {
            val data:ArticleDetailBean?
            try {
                data=call.execute().body()
                launch (UI){
                    detailView.loadWebDetail(data!!)

                }


            }catch (t: IOException){
                t.printStackTrace()
            }

        }
//        call.enqueue(object : Callback<ArticleDetailBean> {
//
//            override fun onResponse(call: Call<ArticleDetailBean>, response: Response<ArticleDetailBean>) {
//
//
//                val data = response.body()
//                detailView.loadWebDetail(data!!)
//
//
//            }
//
//            override fun onFailure(call: Call<ArticleDetailBean>, t: Throwable) {
//                t.printStackTrace()
//            }
//        })
    }
}
